method.rb

```diff
@@ -0,0 +1,181 @@
+module Enumerable # rubocop:disable Style/ModuleLength
+  def my_each
+    i = 0
+    arr = to_a
+    return to_enum(:my_each) unless block_given?
+
+    while i < arr.size
+      yield arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = to_a
+    return to_enum(:my_each_with_index) unless block_given?
+
+    while i < arr.size
+      yield arr[i], i
+      i += 1
+    end
+
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+
+    new_arr
+  end
+
+  def my_all?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = true
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
+        end
+      else
+        my_each do |num|
+          stat = num.to_s.match?(arg.to_s)
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = false unless yield num
+      end
+    else
+      my_each do |num|
+        bool = !num
+        stat = !bool
+      end
+    end
+    stat
+  end
+
+  def my_any?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = false
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
+        end
+      else
+
+        my_each do |num|
+          stat = num.to_s.match?(arg.to_s)
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = true if yield num
+      end
+    else
+      my_each do |num|
+        bool = !num
+        stat = !bool
+      end
+    end
+    stat
+  end
+
+  def my_none?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = true
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
```

Comment:
- [ ] Here you have the opposite issue you have on the my_all? method.
It should return false as soon as you find any item that is from the argument Class.
Ex:
[1,2,3,'work', true].none?(TrueClass)
=> false